theano/sandbox/cuda/dnn_gw.c

```diff
@@ -30,6 +25,21 @@ APPLY_SPECIFIC(conv_gw)(CudaNdarray *input, CudaNdarray *output,
     return 1;
 #endif
 
+
+  if (CudaNdarray_HOST_DIMS(input)[0] == 0) {
+    // zero batch size. Return a zero gradient without calling cuDNN (which will fail on zero batch size)
+    Py_XDECREF(*kerns);
+    *kerns = (CudaNdarray *) CudaNdarray_ZEROS(CudaNdarray_NDIM((*kerns)), (int *) CudaNdarray_HOST_DIMS((*kerns)));
+    if (*kerns == NULL) return 1;
+    Py_INCREF(*kerns);
```

Comment:
Aha, OK I'll remove the INCREF.

The nearest I saw to a memset method was `CudaNdarray_setitem` which can be called with "a python scalar with value 0 when the left side part is c contiguous", but fails otherwise. If I have to have an alternative branch anyway for the case where the existing in-place array isn't C-contiguous then I thought easier just to return a fresh array